src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -15927,6 +15997,18 @@ public class C4<T2> : I0<(int a, int b)>, I0<(T2 a, T2 b)> { }
         }
 
         [Fact]
+        public void InterfaceUnification2()
```

Comment:
>InterfaceUnification2 [](start = 20, length = 21)

It would probably be good to have a test to verify that unification is properly detected when tuple types are involved.